iml-agent/src/action_plugins/ltuer.rs

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2019 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+use crate::agent_error::ImlAgentError;
+use std::path::{Path, PathBuf};
+use tokio::io::{AsyncWrite, AsyncWriteExt};
+use tokio_fs as fs;
```

Comment:
Is there any reason to prefer to`kio::fs` over `std::fs`? I can see that you use asynchronous functions always with `.await`.

Reply:
I much as I understand, only `write_ltuer_conf` is required to be async, Everything else inside it wuold be simpler if synchronous.

Reply:
Everything in iml-agent is required to be async.

There are many tasks that need to make progress over a finite number of cores. If any one call is blocking then it doesn't move into an idle state when possible and allow another task to run instead. This reduces the amount of concurrency possible.

In contrast the async version (tokio_fs in this case) will move into an idle state when it's blocked on an external resource, freeing up to allow another task to be moved forward.

This is a very important invariant as we scale the number of things happening at once through the use of action plugins and daemon plugins.

Reply:
Ok, I think I have confused `.await` with `block_on` :)